### PR TITLE
Fix for #1863

### LINKED ISSUE #1863: [OpenTX2.x] [Companion] Log viewer issue
There is a graph scaling issue when displaying more than 2 cells.
Please check the snapshots below.
Another comment, the X axis should be displayed a Second (Seconde), not Hours (Heure)
log1
log2

### PR DESCRIPTION
**Pull request by griebd**
Fixes #1863, and a small bug when you select some axis to zoom and them deselect. It won go back to zooming both axis after deselecting x and y.

### CODE REVIEW

**Comment by bsongis** [comment]
Thanks for your help, the code you added seems nice compared to the one you removed. However I never went in this code before, I would prefer Damjan press the Merge button as I think he already had a look.

**Comment by projectkk2glider** [comment]
Ok, I will have a look, later...

File: companion/src/logsdialog.cpp [hunk 1]
```diff
@@ -44,18 +44,20 @@ logsDialog::logsDialog(QWidget *parent) :
...
- 
+
   // make bottom and left axes transfer their ranges to top and right axes:
   connect(ui->customPlot->xAxis, SIGNAL(rangeChanged(QCPRange)), ui->customPlot->xAxis2, SLOT(setRange(QCPRange)));
-  connect(ui->customPlot->yAxis, SIGNAL(rangeChanged(QCPRange)), ui->customPlot->yAxis2, SLOT(setRange(QCPRange)));
-  
+  // connect(ui->customPlot->yAxis, SIGNAL(rangeChanged(QCPRange)), ui->customPlot->yAxis2, SLOT(setRange(QCPRange)));
+  connect(ui->customPlot->yAxis, SIGNAL(rangeChanged(QCPRange)), this,
+    SLOT(setRangeyAxis2(QCPRange)));
+
```
Comment (projectkk2glider) [inline companion/src/logsdialog.cpp]: Remove commented line and put your code all on one line (like lines before it).

File: companion/src/logsdialog.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@ logsDialog::logsDialog(QWidget *parent) :
   this->setWindowIcon(CompanionIcon("logs.png"));
   palette.clear();
```
Comment (projectkk2glider) [inline companion/src/logsdialog.cpp]: Since you have defined new colors for grapsh, remove pallete member from logsDialog class.

**Comment by projectkk2glider** [comment]
Found Companion crash, to reproduce:
* open Logs and load one file
* left click on A1 (or any other item)
* then Ctrl + left click on A1 (or the same item as in previous step) -> Crash

**Comment by projectkk2glider** [comment]
Same crash, different way to reproduce:
* open Logs dialog
* load one log
* load another log -> Crash

**Comment by projectkk2glider** [comment]
@griebd I like your changes, you just have to address a couple of things noted above.

**Comment by griebd** [comment]
@projectkk2glider, that is exactly the point of code review... :wink:
About the line breaks, I'm used to limit everything to 80 columns (I hate when I need to use horizontal scroll)... but I have no problems adhering to others project options!!

**Comment by griebd** [comment]
@projectkk2glider, about the crash, tried and no crash on my linux box :disappointed:
do you use windows?

**Comment by jtaylor2** [comment]
I don't get the crash on either Linux or Windows (compiled with Cygwin) - tested after the 9c73f0b commit.  Perhaps it only happens with certain log files, mine are all fairly small.

**Comment by projectkk2glider** [comment]
@griebd  your fixes are good. I will have to research a bit more about my crashes, I am using Linux.

**Comment by griebd** [comment]
which Qt version are you using? mine is 4.8.5.

**Comment by projectkk2glider** [comment]
I found your problem, you are using **release** build of Companion. Try it with the **debug** build, the assertion is only reported there.


```
$ dpkg -l | grep libqt4-dev
ii  libqt4-dev                                  4:4.8.5+git192-g085f851+dfsg-2ubuntu4               amd64        Qt 4 development files
ii  libqt4-dev-bin                              4:4.8.5+git192-g085f851+dfsg-2ubuntu4               amd64        Qt 4 development programs
```

Error degbug session:
```
$ gdb ./companion
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./companion...done.

(gdb) r
Starting program: ./opentx/companion/debug/companion 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffde8b2700 (LWP 19392)]
[New Thread 0x7fffdd3fa700 (LWP 19393)]
[New Thread 0x7fffdcbf9700 (LWP 19394)]
[New Thread 0x7fffcfdf9700 (LWP 19395)]
[New Thread 0x7fffce70a700 (LWP 19396)]
[New Thread 0x7fffcdf09700 (LWP 19397)]
[New Thread 0x7fffb4103700 (LWP 19398)]
[New Thread 0x7fffb3902700 (LWP 19399)]
[New Thread 0x7fffb3101700 (LWP 19400)]
[New Thread 0x7fffb2900700 (LWP 19401)]
[Thread 0x7fffb3101700 (LWP 19400) exited]
[Thread 0x7fffb3902700 (LWP 19399) exited]
[Thread 0x7fffb2900700 (LWP 19401) exited]
[New Thread 0x7fffb2900700 (LWP 19402)]
[New Thread 0x7fffb3902700 (LWP 19403)]
[New Thread 0x7fffb3101700 (LWP 19404)]
[Thread 0x7fffb3902700 (LWP 19403) exited]
[Thread 0x7fffb2900700 (LWP 19402) exited]
[Thread 0x7fffb3101700 (LWP 19404) exited]
ASSERT: "idx >= 0 && idx < s" in file /usr/include/qt4/QtCore/qvarlengtharray.h, line 110

Program received signal SIGABRT, Aborted.
0x00007ffff53dacc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56  ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.

(gdb) bt
#0  0x00007ffff53dacc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007ffff53de0d8 in __GI_abort () at abort.c:89
#2  0x00007ffff6725c92 in qt_message_output(QtMsgType, char const*) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#3  0x00007ffff6725ff9 in ?? () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#4  0x00007ffff6726804 in qFatal(char const*, ...) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#5  0x00000000005ab07d in QVarLengthArray<coords, 256>::operator[] (this=0x7fffffff9570, idx=0)
    at /usr/include/qt4/QtCore/qvarlengtharray.h:110
#6  0x00000000005aa2e6 in QVarLengthArray<coords, 256>::at (this=0x7fffffff9570, idx=0)
    at /usr/include/qt4/QtCore/qvarlengtharray.h:113
#7  0x00000000005a640e in logsDialog::plotLogs (this=0x1269110)
    at ./opentx/companion/src/logsdialog.cpp:667


(gdb) frame 7
#7  0x00000000005a640e in logsDialog::plotLogs (this=0x131d600)
    at ./opentx/companion/src/logsdialog.cpp:667
667   plots.rangeOneMin = plots.coords.at(0).min_y;

(gdb) info locals
plots = {coords = {a = 256, s = 0, ptr = 0x7fffffff9580, {
      array = "\360\226\377\377\377\177\000\000S\317\063\364\377\177\000\000\260<2\364\377\177\000\000\001\000\000\000\000\000\000\000p\360\375\000\000\000\000\000\261)\315\364\377\177\000\000P\227\377\377\377\177\000\000\062S2\364\377\177\000\000\000\000\000\000\000\000\000\000\230\000\000\000\000\000\000\000\r\000\000@\000\000\000\000i\005\310\364\377\177\000\000\r\000\000@\000\000\000\000P\016N\001\000\000\000\000\310\360\375\000\000\000\000\000\340\360\375\000\000\000\000\000\340\360\375", '\000' <repeats 13 times>, "\310\360\375\000\000\000\000\000\261)\315\364\377\177\000\000\340\360\375\000\000\000\000\000\213n3\364\377\177\000\000\310\360\375\000\000\000\000\000\300\226\377\377\377\177\000\000P\227\377\377\377\177\000\000"..., 
      q_for_alignment_1 = 140737488328432, q_for_alignment_2 = 6,9533558065061654e-310}}, min_x = 1427818268, max_x = 0, 
  twoRanges = 21, tooManyRanges = 5, rangeOneMin = 9,9035735385639541e-317, rangeOneMax = 2,1219957904712067e-314, 
  rangeTwoMin = 0, rangeTwoMax = 6,9533481197221767e-310}
hasLogSelection = false

(gdb) p plots.coords 
$1 = {a = 256, s = 0, ptr = 0x7fffffff9580, {
    array = "\360\226\377\377\377\177\000\000S\317\063\364\377\177\000\000\260<2\364\377\177\000\000\001\000\000\000\000\000\000\000p\360\375\000\000\000\000\000\261)\315\364\377\177\000\000P\227\377\377\377\177\000\000\062S2\364\377\177\000\000\000\000\000\000\000\000\000\000\230\000\000\000\000\000\000\000\r\000\000@\000\000\000\000i\005\310\364\377\177\000\000\r\000\000@\000\000\000\000P\016N\001\000\000\000\000\310\360\375\000\000\000\000\000\340\360\375\000\000\000\000\000\340\360\375", '\000' <repeats 13 times>, "\310\360\375\000\000\000\000\000\261)\315\364\377\177\000\000\340\360\375\000\000\000\000\000\213n3\364\377\177\000\000\310\360\375\000\000\000\000\000\300\226\377\377\377\177\000\000P\227\377\377\377\177\000\000"..., 
    q_for_alignment_1 = 140737488328432, q_for_alignment_2 = 6,9533558065061654e-310}}
(gdb) p plots.coords.s
$2 = 0

(gdb) p plots.coords.at(0)
ASSERT: "idx >= 0 && idx < s" in file /usr/include/qt4/QtCore/qvarlengtharray.h, line 110
[Thread 0x7fffdcbf9700 (LWP 19394) exited]

Program received signal SIGABRT, Aborted.
0x00007ffff53dacc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56  ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
The program being debugged was signaled while in a function called from GDB.
GDB remains in the frame where the signal was received.
To change this behavior use "set unwindonsignal on".
Evaluation of the expression containing the function
(QVarLengthArray<coords, 256>::at(int) const) will be abandoned.
When the function is done executing, GDB will silently stop.
(gdb) 
```

The problem is in this sentence:
```c
plots.rangeOneMin = plots.coords.at(0).min_y;
```
The `plots.coords` array is **empty**!

**Comment by griebd** [comment]
Ooopppssss... missed a test and was trying to plot with no field selected, fixed!

**Comment by projectkk2glider** [comment]
@griebd Sorry for the delay, It is good now, merged. But this also needs to be ported to the `next` branch. Can you create a new pull request for that?

**Comment by griebd** [comment]
Will do later. I'm working on the performance issue, and depending on the results I pull it on next including this. Otherwise I make a port of this to next!!!!

**Pull request merged**